charmcraft/commands/extensions.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Infrastructure for the 'extensions' command."""
+import argparse
+from textwrap import dedent
+from typing import Dict
+
+import tabulate
+import yaml
+from craft_cli import emit
+from overrides import overrides
+
+from charmcraft import extensions
+from charmcraft.cmdbase import BaseCommand
+from charmcraft.models.extension import ExtensionModel
+
+_overview = """
+Extensions allow user use pre-defined configuration in their charmcraft.yaml.
+"""
+
+
+class ListExtensionsCommand(BaseCommand):
+    """List available extensions for all supported bases."""
+
+    common = True
+    name = "list-extensions"
```

Comment:
Please include a hidden `extensions` version of this command (see the plurals section at https://discourse.ubuntu.com/t/command-structure/18556).